kendraio/kendraio-app
# Angular upgrade v18

Here is the upgrade to angular 18.
Issue ref: #476 - In the issue, it can be found a good list of the commands used to upgrade.

HOW TO TEST IT:
- You will need node v.18+ to test this branch
- after checking this branch out, remember to update node packages `npm i`
- If you get any error of incompatible version or packages to be updated, I would suggest clearing the cache of NPM packages or even deleting package-lock.json (last resource), if errors persist.
- You can check the preview of this branch here: https://kendraio-app-git-angular-upgrade-18-kendraio.vercel.app/
- Below you can see the main visual difference between v.16 and v.18 Everything else should be the same.


What is missing:
- [x] Fix Ag Grid / Issue with custom components
- [x] Fix small details of UI
- [x] Fix/Double check Forms 
- [x] Leaflet issue
- [ ] Material UI not apply to widget Array Type  

See screenshots:
AgGrid do not display custom components and do not show any error
v18
![image](https://github.com/user-attachments/assets/07aed30b-071c-4141-bd6f-630e58a2711d)

v15
![image](https://github.com/user-attachments/assets/f9e7b206-610e-4ba1-956b-cf3ba1a056ec)


Material UI padding is slightly wrong:
v18
![image](https://github.com/user-attachments/assets/e3cef0da-3a78-411e-ad15-089af2b5b93e)

v15
![image](https://github.com/user-attachments/assets/167d6487-d390-461a-bad6-fafa42c4ae49)


Forms (or material UI?)
v18
![image](https://github.com/user-attachments/assets/300f0324-60f4-47af-a61c-62aa690ff910)

v15
![image](https://github.com/user-attachments/assets/74e5303e-d217-48be-8b9c-c498edaf5fde)

Leaflet issue:
v18
![image](https://github.com/user-attachments/assets/cf1c4c21-0c61-4e7c-a9ff-8105bf23667f)

Widget Form Issue:
v18
![image](https://github.com/user-attachments/assets/b4d110b4-442f-46f2-a1ea-90c5ae11561a)
